qcfractal/cli/qcfractal_server.py

```diff
@@ -3,80 +3,188 @@
+
+    ### Start subcommands
+    start = subparsers.add_parser('start', help="Starts a QCFractal server instance.")
+    start.add_argument("--base-folder", **FractalConfig.help_info("base_folder"))
 
-    manager = parser.add_argument_group('Local QueueManager Settings (optional)')
-    # This option defaults to None if option not present, -1 if present, or value if provided
-    manager.add_argument(
-        '--local-manager',
-        const=-1,
-        default=None,
-        action='store',
-        nargs='?',
-        type=int,
-        help='Creates a local pool QueueManager')
-
-    manager = parser.add_argument_group('Manager Settings')
-    manager.add_argument("--heartbeat-frequency", type=int, default=1800, help="The manager heartbeat frequency.")
-
-    security = parser.add_argument_group('Security Settings')
-    security.add_argument(
-        "--security", type=str, default=None, choices=[None, "local"], help="The security protocol to use")
-    security.add_argument(
-        "--allow-read", type=bool, default=True, help="Allow read-only queries or not if security is active.")
-    security.add_argument("--disable-ssl", type=bool, default=False, help="Disables SSL if present, if False a SSL cert will be created for you")
-    security.add_argument("--tls-cert", type=str, default=None, help="Certificate file for TLS (in PEM format)")
-    security.add_argument("--tls-key", type=str, default=None, help="Private key file for TLS (in PEM format)")
-
-    general = parser.add_argument_group('General Settings')
-    general.add_argument("database_name", type=str, help="The name of the database to use with the storage socket")
-    general.add_argument("--server-name", type=str, default="QCFractal Server", help="The server name to broadcast")
-    general.add_argument("--query-limit", type=int, default=1000, help="The maximum query size to server")
-    general.add_argument("--log-prefix", type=str, default=None, help="The logfile prefix to use")
-    general.add_argument("--database-uri", type=str, default="mongodb://localhost", help="The database URI to use")
-    general.add_argument("--port", type=int, default=7777, help="The server port")
-    general.add_argument("--compress-response", type=bool, default=True, help="Compress the response or not")
-    general.add_argument("--config-file", type=str, default=None, help="A configuration file to use")
-    general.add_argument("--start-periodics", type=bool, default=True, help="Start the periodic calls or not, always recommended unless running fractal-server behind a proxy")
-
-    parser._action_groups.reverse()
+    fractal_args = start.add_argument_group('Server Settings')
+    # Allow port and logfile to be altered on the fly
+    for field in ["port", "logfile"]:
+        cli_name = "--" + field.replace("_", "-")
+        fractal_args.add_argument(cli_name, **FractalServerSettings.help_info(field))
 
+    fractal_args.add_argument("--database-name",
```

Comment:
I tried to keep those classes equivalent to the configuration file itself. This argument allows you to override defaults and create a new blank database. In general, this will be `None`.

Not sure if this is the best way, certainly worth thinking about.